File: src/HotelBooking.Domain/Entities/Room.cs
using HotelBooking.Domain.Common;

namespace HotelBooking.Domain.Entities;

public class Room : BaseEntity
{
    public Guid HotelId { get; set; }
    public Hotel Hotel { get; set; } = default!;

    public Guid CategoryId { get; set; }
    public RoomCategory Category { get; set; } = default!;

    public decimal PricePerNight { get; set; }
    public int Capacity { get; set; }
    public bool IsAvailable { get; set; } = true;

    public ICollection<Booking>? Bookings { get; set; }
}

File: web/js/core/models/room.js
export class Room {
    constructor({ id, number, capacity, pricePerNight, hotelId, hotelName, roomCategoryName }) {
        this.id = id;
        this.number = number;
        this.capacity = capacity;
        this.pricePerNight = pricePerNight;
        this.hotelId = hotelId;
        this.hotelName = hotelName;
        this.roomCategoryName = roomCategoryName;
    }
}
